test/k4FWCoreTest/src/components/ExampleFunctionalConsumer.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2014-2023 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Gaudi/Property.h"
+#include "GaudiAlg/Consumer.h"
+
+#include "edm4hep/MCParticleCollection.h"
+
+// Define BaseClass_t
+#include "k4FWCore/BaseClass.h"
+
+#include <stdexcept>
+#include <string>
+
+// Which type of collection we are reading
+using colltype = edm4hep::MCParticleCollection;
```

Comment:
Can we remove this typedef? It's effectively used twice (as far as I can see), and it reduces readability IMHO. Additionally, there are other examples with multiple collections that show the usage with a typedef which are much clearer, I think.

If not removing it, at least we should give it a more descriptive name.